### ISSUE TITLE
make install doesn't load firmware if board already connected

### ISSUE THREAD

**Issue by JHentges** [accesio/aiousb-alpha#3]
If a USB board sans-firmware (Power/Activity LED is dark) is connected when `make install` is executed the firmware does not get loaded.
Subsequently hotplugging does load the firmware.
This plausibly means firmware won't be loaded into connected boards at power-up; TBD

**Comment by JHentges** [comment]
If the firmware gets loaded into already-connected boards sans-firmware on power-up then this is an `improvement` not a `bug`

**Comment by JHentges** [comment]
I’ve confirmed the firmware *does* get installed following a cold-boot, so this isn’t a priority

**Comment by jdolanIV** [comment]
I think this should be closed as expected behavior. If you really want to avoid a reboot or a hotplug then you could manually insmod the driver after the build. We could maybe add a step that insmods the driver after the build, but I'd rather not.

**Comment by JHentges** [comment]
Doesn’t the “make install” do an insmod?
I figured it was a matter of needing to fire a synthetic hotplug event or something

— John.Hentges [@accesio.com]

**Comment by jdolanIV** [comment]
No. It copies the .ko file to the correct location and then runs depmod. The depmod command sets up the hotplugging so it will work, but if the device is already there then no hotplug will be generated for it to pick up. A quick Google search turns up "udevadm trigger". If that works we could try to add it to the install for the module.

**Comment by JHentges** [comment]
Excellent.  We definitely want to avoid hotplug and reboot being necessary to get a board working if we can.

— John.Hentges [@accesio.com]
